src/faebryk/libs/sexp/dataclass_sexp.py

```diff
@@ -275,31 +281,36 @@ def _decode[T](
                 value_dict[name] = out
 
     # Positional
-    for f, v in (it := zip_non_locked(positional_fields.values(), pos_values.values())):
+    for f, (sexp_i, v) in (
+        it := zip_non_locked(positional_fields.values(), pos_values.items())
+    ):
         sp = sexp_field.from_field(f)
         # special case for missing positional empty StrEnum fields
-        if isinstance(f.type, type) and issubclass(f.type, StrEnum):
-            if "" in f.type and not isinstance(v, Symbol):
-                value_dict[f.name] = _convert(Symbol(""), f.type, stack, f.name, sp)
-                # only advance field iterator
-                # if no more positional fields, there shouldn't be any more values
-                if it.next(0) is None:
-                    raise ValueError(f"Unexpected symbol {v}")
-                continue
-
-        # positional list = var args
-        origin = get_origin(f.type)
-        if origin is list:
-            vs = []
-            next_val = v
-            # consume all values
-            while next_val is not None:
-                vs.append(next_val)
-                next_val = it.next(1, None)
-            out = _convert(vs, f.type, stack, f.name, sp)
+        if (
```

Comment:
This is a simple `continue` -> `else` replacement so there's only one loop exit point